### ISSUE TITLE
FlexForm not shown if using sub-directory inside Templates/Content/...

### ISSUE THREAD

**Issue by typoworx-de** [FluidTYPO3/fluidcontent#323]
FlexForm is not showing-up if using sub-directory inside Templates/Content/.

I've tried to have some bootstrap-grid and flexforms inside this folder of my template-extension:
/Resources/Private/Templates/Content/Bootstrap/

After about 3 hours of research I figured out that this DOES NOT work indeed of that the templates are correctly found and showing in Content-Elements Wizard and further selection.

Just the Flexform (flux:form) is completly missing and replaced with some kind of "default flexform" just showing up like this:

```
Plugin Options

  DEF:
The Title:
[ LONG TEXT INPUT FIELD ]
``´

This should have been mentioned somewhere!

Is it possible to fix that so that FlexForm configuration is rendered correctly? I think there is some path-mismatch that does for some reason not provide the flux:form configuration and/or Sections/Fields from the Flux-Template!

**Comment by NamelessCoder** [comment]
Subfolders are not supported *unless* you add each sub-folder as individual template paths. This is due to the way references to template files get stored and resolved.

You should think of these templates as *controller actions*. Meaning that `Content` is your controller name and any template you put in the `$templateRootPath/Content/` folder must fit within the Controller-and-action convention. Also note that Flux and other FluidTYPO3 packages do not support sub-packages (which are somewhat redundant).

Although the documentation doesn't explicitly say that you cannot use subfolders, it does explain that the Controller-and-action conventions apply and only uses correct locations for examples. The alternative would be to explain everything you *can't* do which is infinitely harder and more complex to do than to explain what you can and should do.

**Comment by unrevised6419** [comment]
@NamelessCoder I also tried somehow to group Flux Content Elements, because folder is gettings pretty big.  
I tried this but it didn't work.  

Added new controller that extends `ContentController`
```bash
Controllers/
  PageController.php
  ContentController.php
  AnotherContentController.php # new controller
```
Created new folder for Flux Content Elements
```bash
Templates/
  Page
  Content
  AnotherContent # new folder for elements
```
Registered controller `ext_localconf.php`
```php
\FluidTYPO3\Flux\Core::registerProviderExtensionKey('Vendor.ExtensionKey', 'Page');
\FluidTYPO3\Flux\Core::registerProviderExtensionKey('Vendor.ExtensionKey', 'Content');
\FluidTYPO3\Flux\Core::registerProviderExtensionKey('Vendor.ExtensionKey', 'AnotherContent');
```

But elements inside `AnotherContent` folder does not appear in backend.  
Is this the right way or not?!  
Thanks!

**Comment by unrevised6419** [comment]
Realised that commented in `fluidcontent` but I am using only `flux`

**Comment by NamelessCoder** [comment]
@iamandrewluca The issue has also been closed for nearly 2.5 years - normally I don't reply to messages in long-closed issues but in this case: please create a new issue in the Flux repository as a feature request for nested folders in the templateRootPath you define.

The MVC allows sub-controllers, Flux can too - though it currently does not. You do need to register the `Content` controller name but subfolders could become supported.

Will lock the issue from further comments. There's no need to apologise for commenting on the dead issue - no harm done!
